src/service/git/octokit.ts

```diff
@@ -14,9 +15,11 @@ export class OctokitFactory {
    */
   public static getOctokitInstance(): Octokit {
     if (!this.octokit) {
+      const proxy = process.env["http_proxy"];
```

Comment:
this seems weird too... :thinking: please explain

Reply:
Same, it is part of the limitation. Behind the scenes `act-js` uses proxies to mock apis. It expects that all clients respect the `http_proworkaroundxy` env variable. Most of the clients do but there are some which don't and in our case Octokit doesn't. Their reasoning being that people might use Octokit in browsers which doesn't follow this convention see [here](https://github.com/octokit/octokit.js/issues/2098#issuecomment-844673894)

So I ended up adding a workaround to force Octokit to respect it which is what this line of code does. Now I was debating whether I should wrap this around `ACT` env variable and make proxies available only during testing. But then I thought if for example `build-chain` is used in an enterprise setting, it might need to go through proxies and this setting will help with that. However, do let me know if that is a bit far fetched. I can wrap it around the `ACT` env var

Reply:
it would be nice to add a comment briefly explaining why, don't you think so?

Reply:
Will do!